bazel/p4c_deps.bzl

```diff
@@ -0,0 +1,52 @@
+"""Load dependencies needed to compile p4c as a 3rd-party consumer."""
+
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+def p4c_deps():
+    """Loads dependencies need to compile p4c."""
+    if not native.existing_rule("com_github_nelhage_rules_boost"):
+        git_repository(
+            name = "com_github_nelhage_rules_boost",
+            commit = "1da7517245fb944d6b7b427aa86fd5571663f90a",
+            remote = "https://github.com/nelhage/rules_boost",
+            shallow_since = "1591812253 -0700",
+        )
+    if not native.existing_rule("com_github_p4lang_p4runtime"):
+        # Cannot currently use local_repository due to Bazel limitation,
+        # see https://github.com/bazelbuild/bazel/issues/11573.
+        #
+        # native.local_repository(
+        #     name = "com_github_p4lang_p4runtime",
+        #     path = "@com_github_p4lang_p4c//:control-plane/p4runtime/proto",
+        # )
+        #
+        # We use git_repository as a workaround; the version used here should
+        # ideally be kept in sync with the submodule control-plane/p4runtime.
+        git_repository(
+            name = "com_github_p4lang_p4runtime",
+            remote = "https://github.com/p4lang/p4runtime",
+            commit = "776797d4bc7c2e5f1a7249f73f1c879a91688e75",
```

Comment:
So we have to edit this manually every time we make a commit to p4runtime?

Reply:
Yes and no. We can refer to any p4runtime commit here, it doesn't necessarily have to be the latest one. So we do not have to update this every time we make a commit to p4runtime.
This works much like git submodules, which p4c is already using for the cmake-based build. As explained in the comment, we would ideally have this rule point at the submodule, so that cmake and Bazel would be guaranteed to use the same version of p4runtime, but due to a Bazel limitation we cannot currently do so. The good news is that if something brakes, CI will tell us.